File: src/main/java/de/huddeldaddel/euler/math/FibonacciSequence.java
package de.huddeldaddel.euler.math;

import java.math.BigInteger;

public class FibonacciSequence {

    private BigInteger num1 = BigInteger.ZERO;
    private BigInteger num2 = BigInteger.ZERO;

    public BigInteger nextValue() {
        if(BigInteger.ZERO.equals(num1))
            return num1 = BigInteger.ONE;

        if(BigInteger.ZERO.equals(num2))
            return num2 = BigInteger.ONE;

        final BigInteger result = num1.add(num2);
        num1 = num2;
        return num2 = result;
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/sequences/FibonacciSequence.kt
package de.huddeldaddel.euler.sequences

import java.math.BigInteger

class FibonacciSequence {

    private var num1 = BigInteger.ZERO
    private var num2 = BigInteger.ZERO

    fun nextValue(): BigInteger {
        if (BigInteger.ZERO == num1) {
            num1 = BigInteger.ONE
            return num1
        }

        if (BigInteger.ZERO == num2) {
            num2 = BigInteger.ONE
            return num2
        }


        val result = num1.add(num2)
        num1 = num2
        num2 = result
        return num2
    }

}
